### PR TITLE
Fix navigation in mobile devices

### LINKED ISSUE #627: Mobile course navigation elements are inside the main navigation
In the DOM order, some elements which are usually in the "Course Navigation" are inside the "Main Navigation" at certain screen sizes.

This breaks the landmarks and logical order of the page content.

It is possible that this will be too big a task to fully resolve this summer (it's potentially going to need a new template structure), but I'll do some digging anyway.

### PR DESCRIPTION
**Pull request by jaguarfi**
# Description

In the issue #627 it was reported that the use of the toggle button for mobile devices was breaking the landmarks and logical of the page content. In addition, some other small issues were addressed in this PR.

* The button "skip navigation" of the left side course menu was present in small devices and it was overlapping other elements inside the toggle menu. Therefore, that specific "skip navigation" link was removed in small devices. It is not necessary to have a "skip navigation" button inside the toggle menu.
* The tab navigation order of the navbar elements in mobile devices was in the wrong order. Therefore, the elements were arranged to follow a logical flow from left to right.
* The focus styles were not applied to the toggle button. Therefore, the styles were applied to the toggle button. One for keyboard navigation and other for the click/touch interaction.
* The contrast of the links located inside the toggle menu was low. Therefore, the `color` property was changed.
* The brand button was not interactive in small devices. Therefore, the brand button interaction was added to small devices.

## Existing
old_navigation

## NEW Changes
new_navigation

Fixes #627

# Testing

The changes were testes manually.

# Have you updated the README or other relevant documentation?

...

# Is it [Done](https://wiki.aalto.fi/display/EDIT/Definition+of+Done)?

*Clean up your git commit history before submitting the pull request!*

- [ ] I (developer) have created unit tests and the tests pass
- [ ] I (developer) have created functional tests (Selenium tests) if applicable
- [X] I (developer) have tested the changes manually
- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

### CODE REVIEW

File: templates/base.html [hunk 1]
```diff
@@ -79,6 +79,10 @@
                 <nav class="topbar navbar navbar-inverse navbar-static-top" aria-label="{% trans "Main" %}">
                     <div class="container-fluid">
                         <div class="navbar-header">
+							<a class="navbar-brand" href="{% url 'home' %}">{% brand_name %}</a>
```
Comment (Mankro) [inline templates/base.html:82]: What does this change do? Would you clarify in the discussion here?

File: templates/base.html [hunk 2]
```diff
@@ -89,10 +93,6 @@
                                 <span class="icon-bar"></span>
                                 <span class="icon-bar"></span>
                             </button>
-                            <a class="navbar-brand hidden-xs" href="{% url 'home' %}">{% brand_name %}</a>
-                            <span class="navbar-brand visible-xs">
-                            {% brand_name %}{% if instance %} &nbsp; {{ course.name|parse_localization }}{% endif %}
-                            </span>
                         </div>
                         <div class="collapse navbar-collapse" id="bs-navbar-collapse">
                             <ul class="nav navbar-nav hidden-xs">
```
Comment (Mankro) [inline templates/base.html:98]: I wonder if the course instance selection menu could be shown in the mobile view too. Could you try this?

**Review by Mankro — CHANGES_REQUESTED** [review]
Nice! There are some issues to fix.

* very low contrast: mobile menu, focus during keyboard tabbing, yellow background and white font. Font color must be changed in the focus style.
* very low contrast: gray "course staff" text (not link) in the mobile menu. Also very small font size warning. The same for text "Course" and "Site" in the mobile menu.
* "Course" text is not well aligned with the language button
* The language selection menu (Finnish/English) does not look good in the mobile menu. You must fix the background and font colours. You can set language "|en|fi|" in the A+ course settings even if the course is only defined in one language like the aplus-manual.

File: course/templates/course/_course_menu.html [hunk 3]
```diff
@@ -8,7 +8,7 @@
 {% prepare_course_menu %}
 
 <li>
-	<a href="#course-content" class="skip-link page-skip-link">
+	<a href="#course-content" class="skip-link page-skip-link hidden-xs">
```
[the remark this answers is not in the document]
Comment (Mankro) [reply course/templates/course/_course_menu.html:11]: The language menu must be also tested with the screen reader.

[the remark this answers is not in the document]
Comment (jaguarfi) [reply course/templates/course/_course_menu.html:11]: It seems like mobile devices have certain features that allow users to navigate websites with certain gestures.
https://developer.mozilla.org/en-US/docs/Learn/Accessibility/Mobile. But, I was also checking some websites, and they keep the main "skip navigation" link in small devices. You can check in bootstrap, GOV.UK and GitHub websites. 

> Can you verify how screen readers behave with the mobile course menu? What does it say on top of the mobile menu open button?

It says "toggle navigation button collapsed" if the menu is closed or "toggle navigation button open" if it is open. I think that is the expected behaviour, or at least that message is used also in bootstrap and Github webpages.

[the remark this answers is not in the document]
Comment (jaguarfi) [reply course/templates/course/_course_menu.html:11]: > The language menu must also be tested with the screen reader.

Which language menu? This?
image
If so, I tested the behaviour of that button in production, and it works as intended, but we can take a closer look at that button when solving this issue https://github.com/apluslms/a-plus/issues/428

File: templates/base.html (hunk 1 above)
Comment (jaguarfi) [reply templates/base.html:82]: The brand button had an `anchor` pointing to the home page, but the `anchor` is removed for small devices, and that makes no sense. Users in small devices also expect to brand button to behave as a link to the home page. That is the reason I remove the code below. The code that is shown in green already existed, but the changes in indentation makes the code looks like a new piece of code. I only replaced the `hidden-xs` class with the `visible-xs` class, and move it the **brand** button before the toggle menu.

File: templates/base.html (hunk 2 above)
Comment (jaguarfi) [reply templates/base.html:98]: I think we can keep the `drop-down menu` in the `topbar`.

Comment (jaguarfi) [reply templates/base.html:98]: What about this?

image


Or we can keep the dropdown menu next to the brand icon just as in the desktop version. 
image
@Mankro which one do you like more?

File: course/templates/course/_course_menu.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mankro) [reply course/templates/course/_course_menu.html:11]: Yes, I mean the language menu that selects between Finnish and English.

File: templates/base.html (hunk 2 above)
Comment (Mankro) [reply templates/base.html:98]: I like the second one (separate course instance menu and the toggle menu like in the desktop version).

**Comment by jaguarfi** [comment]
I applied the requested changes

**Comment by jaguarfi** [comment]
@Mankro I have one commit that format the base.html file, so I think you will have to review the changes by commit :)

**Comment by jaguarfi** [comment]
There is something  I need to fix!!

**Comment by jaguarfi** [comment]
Now is ready 👍🏽

**Review by Mankro — APPROVED** [review]
Excellent work! I will make a new issue about too low contrast in the new notification dropdown of the mobile course menu. Let's merge this now.

**Comment by Mankro** [comment]
@jaguarfi new small issue about low contrast: https://github.com/apluslms/a-plus/issues/723

**Pull request merged**